service-front/module/Application/test/Controller/DonorFlowControllerTest.php

```diff
@@ -234,6 +234,78 @@ public function testDonorIdMatchPage(): void
     }
 
 
+    public function testWhatIsVouchingPageOptNo(): void
```

Comment:
have added 3 separate tests for the what-is-vouching page as it was getting complex using a `dataProvider` but i'm not sure what best practice would be here, should i stick to a single test with a `dataProvider` and conditional logic in the test?

Reply:
No, I think that's good. It's only worth using a dataprovider when you have lots of clearly similar tests, like testing a validation function.